### PR TITLE
[5.0][Filesystem] add parameter type hints

### PR DESCRIPTION
**Pull request by smoench**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #32179
| License       | MIT
| Doc PR        | N/A

This PR adds parameter type hints to the Filesystem component.

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Tests/FilesystemTest.php [hunk 1]
```diff
@@ -1397,7 +1397,6 @@ public function providePathsForIsAbsolutePath()
             ['var/lib', false],
             ['../var/lib', false],
             ['', false],
-            [null, false],
```
Comment (derrabus) [inline src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: Why do we have to remove this one?

**Comment by smoench** [comment]
Should `FilesystemTest::testChmodWithWrongModLeavesPreviousPermissionsUntouched` be removed as it is not possible anymore passing a mode parameter as a string to `Filesystem::chmod`?

File: src/Symfony/Component/Filesystem/Tests/FilesystemTest.php (hunk 1 above)
Comment (smoench) [reply src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: `Filesystem::isAbsolutePath` accepts only strings.

Comment (nicolas-grekas) [reply src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: then we should deprecate passing `nulll` in 4.4 first

Comment (smoench) [reply src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: Okay, I will provide a PR

Comment (smoench) [reply src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: https://github.com/symfony/symfony/pull/32315

**Review by fabpot — CHANGES_REQUESTED** [review]
As done in similar PRs for other components, can you removed the phpdocs that are not needed anymore?

**Comment by smoench** [comment]
Not needed phpdocs are removed

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @smoench.

**Pull request merged**